plugins/formal-verification/testData/diagnostics/no_contracts/smartcast.fir.diag.txt

```diff
@@ -0,0 +1,54 @@
+/smartcast.kt:(4,19): info: Generated Viper text for smartcastReturn:
+method global$fun_smartcastReturn$fun_take$NT_Int$return$T_Int(local$n: dom$Nullable[Int])
+  returns (ret$0: Int)
+{
+  inhale dom$Type$isSubtype((dom$TypeOf$typeOf(local$n): dom$Type), dom$Type$special$Nullable(dom$Type$Int()))
+  if (!(local$n == (dom$Nullable$null(): dom$Nullable[Int]) &&
+  (dom$Nullable$null(): dom$Nullable[dom$Unit]) ==
+  (dom$Nullable$null(): dom$Nullable[dom$Unit]) ||
+  local$n != (dom$Nullable$null(): dom$Nullable[Int]) &&
+  (dom$Nullable$null(): dom$Nullable[dom$Unit]) !=
+  (dom$Nullable$null(): dom$Nullable[dom$Unit]) &&
```

Comment:
I've noticed that sometimes we get redundant checks, should we file an issue for this? Or we do not consider it for the moment?

Reply:
The problem is that we compile `x == y` to `(x == null && y == null) || (x != null && y != null && cast(x) == cast(y))`.  Of course, when `y == null` this is unnecessarily complicated.  I don't think it's really a problem for now.